ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -296,8 +296,9 @@ private static SourceName getSourceName(final PlanNode node) {
         final DataSourceType rightType) {
 
       return joinerMap.getOrDefault(new Pair<>(leftType, rightType), () -> {
-        throw new KsqlException("Join between invalid operands requested: left type: "
-            + leftType + ", right type: " + rightType);
+        throw new KsqlException(
```

Comment:
We have a check in the new `verifyJoin` method and thus should never hit this line any longer, ie, the user should never see this error message again.

Reply:
If users will never hit it, whey do we throw it?

Reply:
We could remove it but would hit a NPE (if there is a bug) for this case -- thus, it seems better to just throw if we hit a bug instead of crashing with NPE?

It also seems safer to just keep the exception to void introducing a regression by accident (ie, if we miss a case and user could still hit this, even if they should not).

Not sure how this is handled in ksqlDB normally?